src/plugins/telemetry_collection_manager/server/plugin.ts

```diff
@@ -120,9 +121,10 @@ export class TelemetryCollectionManagerPlugin
       ? this.elasticsearchClient?.asScoped(config.request).asCurrentUser
       : this.elasticsearchClient?.asInternalUser;
     // Scope the saved objects client appropriately and pass to the stats collection config
-    const soClient = config.unencrypted
-      ? this.savedObjectsService?.getScopedClient(config.request)
+    const soRepository = config.unencrypted
+      ? this.savedObjectsService?.createScopedRepository(config.request)
```

Comment:
We're now getting the scoped repository instead of the client straight away. This way we can extend both clients: the scoped and internal with the same class `TelemetrySavedObjectsClient`.